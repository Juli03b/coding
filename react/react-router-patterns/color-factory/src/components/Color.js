import { Link, useParams } from "react-router-dom";
import "../styles/Color.css"

const Color = ({colors}) => {
    const {color} = useParams();

    return (
        <div className="Color" style={{backgroundColor: colors?.[color] || color}}>
            <h1 className="ColorHeader">{color}</h1>
            <Link to="/colors">Go to color picker</Link>
        </div>
    );
}

export default Color;