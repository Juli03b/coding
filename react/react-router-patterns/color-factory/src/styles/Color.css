div.Color {
    height: 63rem;
}
h1.ColorHeader {
    text-transform: capitalize;
}