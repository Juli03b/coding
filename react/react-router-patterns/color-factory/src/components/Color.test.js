import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Color from "./Color";

it("should render", () => {
    render(
        <MemoryRouter>
            <Color />
        </MemoryRouter>
    );
});

it("should match snapshot", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <Color />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot()
});