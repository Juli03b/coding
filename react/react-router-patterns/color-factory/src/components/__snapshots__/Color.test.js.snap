// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Color"
  >
    <h1
      class="ColorHeader"
    />
    <a
      href="/colors"
    >
      Go to color picker
    </a>
  </div>
</DocumentFragment>
`;
